Gifski/DraggableFile.swift

```diff
@@ -0,0 +1,63 @@
+import Cocoa
+
+final class DraggableFile: NSImageView, NSDraggingSource {
+	private var mouseDownEvent: NSEvent!
+
+	var fileUrl: URL! {
+		didSet {
+			image = NSImage(byReferencing: fileUrl)
+		}
+	}
+
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+		isEditable = false
+		unregisterDraggedTypes()
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	func draggingSession(_ session: NSDraggingSession, sourceOperationMaskFor context: NSDraggingContext) -> NSDragOperation {
+		return .copy
+	}
+
+	override func mouseDown(with event: NSEvent) {
+		mouseDownEvent = event
+	}
+
+	override func mouseDragged(with event: NSEvent) {
+		let mouseDownPoint = mouseDownEvent.locationInWindow
+		let dragPoint = event.locationInWindow
+		let dragDistance = hypot(mouseDownPoint.x - dragPoint.x, mouseDownPoint.y - dragPoint.y)
+
+		if dragDistance < 3 {
```

Comment:
I don't understand the point of this? I tried removing it and the dragging feels better.